docs/source/notes/extending.rst

```diff
@@ -854,6 +853,62 @@ useful to confirm that a function that isn't present in the dictionary returned
+hinted by its name, it is responsible to call the right backend
+function for a specific call of a function. For example, when calling
+``torch.add(a, b)``, the dispatcher will inspect both argument, figure out which
+"feature" (autograd, autocast, functionalization, etc) and which "backend" (CPU,
+CUDA, MPS, etc) should be used for this specific call and finally call all the
+right kernels.
+A very common thing done by a kernel is "redispatch". For example when running your
+neural network on GPU with autocast, the call will be first the autocast kernel that
+will handle any potential autocast logic and redispatch down. The next feature in line
+will be autograd that will properly create the autograd graph and redispatch down.
+Finally, we reach the backend kernel for CUDA which will launch the right CUDA kernel
+and return the final result. On the way out, autograd will attach the graph to the
+output and then autocast will have a chance to do any update it needs on exit.
+
+One of the configuration of the dispatcher is the order in which all these feature and backend keys are called. The latest list can be found in ``DispatchKey.h`` inside the ``DispatchKey`` enum. For the purpose of extending torch, the important piece of ordering is: vmap -> Autocast -> Autograd -> ZeroTensor -> Neg/Conj -> Functionalize -> Python -> Backends. The most important key for the purpose of this discussion is the ``Python`` one as every Tensor subclass with the ``__torch_dispatch__`` method defined will have the flag to call into this feature. It is from there that the user-defined method is called and where the behavior can be overwritten arbitrarily. From there, calling the provided ``func`` again will perform a "redispatch".
+
+Some important implication of this implementation are:
+- This code runs "below all features" it is thus only responsible, like a regular backend, to generate the output value of each Tensor (and can, and should, ignore all advanced features like autograd, autocast, etc).
+- If any high level feature implement a give function without redispatching, it will never reach the Python key and so the ``__torch_dispatch__`` callback will never be triggered. This happens in particular for CompositeImplicitAutograd function (composite functions whose autograd formula is specified implicitly by the other native ops it calls) which are evaluated at the Autograd level without redispatching.
+- When calling back to python and when wrapping the results, the usual conversions are used as some objects cannot be represented in python (undefined Tensors for example become None).
+- To have an easy to use python object representing a given native function, they are lazily populated at ``torch.ops.{namespace}.{func_name}.{overload_name}``. The ``func`` object given to ``__torch_dispatch__`` is always an entry from this namespace. This namespace can be used to directly call native ops and bypass the usual python API and binding code.
```

Comment:
```suggestion
- Our native functions are lazily populated as ``torch.ops.{namespace}.{func_name}.{overload_name}`` as Python objects to enable calling them in Python. The ``func`` object given to ``__torch_dispatch__`` is always an entry from this namespace. This namespace can be used to directly call native ops and bypass the usual python API and binding code.
```
Is this the intent of the phrasing?

Reply:
Yes! Thanks for the proposed rephrasing.